pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/ModelInterpreter.groovy

```diff
@@ -138,34 +128,82 @@ class ModelInterpreter implements Serializable {
+     * @param skippedForWhen True if thisStage is already skipped for a failed when condition
+     * @return A map of parallel branch names to closures to pass to the parallel step
+     */
+    def getParallelStages(Root root, Agent parentAgent, Stage thisStage, Throwable firstError, boolean skippedForFailure,
+                          boolean skippedForUnstable, boolean skippedForWhen) {
         def parallelStages = [:]
-        thisStage?.parallel?.stages?.each { parallelStage ->
+        thisStage?.parallelContent?.each { content ->
             if (skippedForFailure) {
-                parallelStages.put(parallelStage.name, {
-                    script.stage(parallelStage.name) {
-                        Utils.logToTaskListener("Stage '${parallelStage.name}' skipped due to earlier failure(s)")
-                        Utils.markStageSkippedForFailure(parallelStage.name)
+                parallelStages.put(content.name, {
+                    script.stage(content.name) {
+                        Utils.logToTaskListener("Stage '${content.name}' skipped due to earlier failure(s)")
+                        Utils.markStageSkippedForFailure(content.name)
+                        if (content.stages != null) {
+                            evaluateStage(root, thisStage.agent ?: parentAgent, content, firstError, thisStage).call()
+                        }
                     }
                 })
             } else if (skippedForUnstable) {
-                parallelStages.put(parallelStage.name, {
-                    script.stage(parallelStage.name) {
-                        Utils.logToTaskListener("Stage '${parallelStage.name}' skipped due to earlier stage(s) marking the build as unstable")
-                        Utils.markStageSkippedForUnstable(parallelStage.name)
+                parallelStages.put(content.name, {
+                    script.stage(content.name) {
+                        Utils.logToTaskListener("Stage '${content.name}' skipped due to earlier stage(s) marking the build as unstable")
+                        Utils.markStageSkippedForUnstable(content.name)
+                        if (content.stages != null) {
+                            evaluateStage(root, thisStage.agent ?: parentAgent, content, firstError, thisStage).call()
+                        }
                     }
                 })
             } else if (skippedForWhen) {
-                parallelStages.put(parallelStage.name, {
-                    script.stage(parallelStage.name) {
-                        Utils.logToTaskListener("Stage '${parallelStage.name}' skipped due to when conditional")
-                        Utils.markStageSkippedForConditional(parallelStage.name)
+                parallelStages.put(content.name, {
+                    script.stage(content.name) {
+                        Utils.logToTaskListener("Stage '${content.name}' skipped due to when conditional")
```

Comment:
could `when` be marked up in some way?

Reply:
I try to avoid markup in the console log.